code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm

```diff
@@ -191,6 +191,12 @@
 /mob/living/simple_animal/drone/cogscarab/ratvar_act()
 	fully_heal(TRUE)
 
+/mob/living/simple_animal/drone/cogscarab/can_speak_in_language(datum/language/dt)
```

Comment:
Oh, you mean move the "can_speak_in_language" check into the language datum itself?